model_server_util_tools/model_packaging/export_model_utils.py

```diff
@@ -0,0 +1,239 @@
+        import mxnet as mx
+        model_name = os.path.splitext(os.path.basename(onnx_file))[0]
+        symbol_file = '%s-symbol.json' % model_name
+        params_file = '%s-0000.params' % model_name
+        signature_file = 'signature.json'
+        # Find input symbol name and shape
+        model_proto = onnx.load(os.path.join(model_path, onnx_file))
+        graph = model_proto.graph
+        _params = set()
+        for tensor_vals in graph.initializer:
+            _params.add(tensor_vals.name)
+
+        input_data = []
+        for graph_input in graph.input:
+            shape = []
+            if graph_input.name not in _params:
+                for val in graph_input.type.tensor_type.shape.dim:
+                    shape.append(val.dim_value)
+                input_data.append((graph_input.name, tuple(shape)))
+
+        sym, arg_params, aux_params = onnx_mxnet.import_model(os.path.join(model_path, onnx_file))
+        # UNION of argument and auxillary parameters
+        params = dict(arg_params, **aux_params)
+        # rewrite input data_name correctly
+        with open(os.path.join(model_path, signature_file), 'r') as f:
+            data = json.loads(f.read())
+            data['inputs'][0]['data_name'] = input_data[0][0]
+        with open(os.path.join(model_path, signature_file), 'w') as f:
+            f.write(json.dumps(data, indent=2))
+
+        with open(os.path.join(model_path, symbol_file), 'w') as f:
+            f.write(sym.tojson())
+
+        save_dict = {('arg:%s' % k): v.as_in_context(mx.cpu()) for k, v in params.items()}
+        mx.nd.save(os.path.join(model_path, params_file), save_dict)
+        return symbol_file, params_file
+
+    @staticmethod
+    def generate_publisher(publisherargs):
+        publisher = Publisher(publisherargs.author, publisherargs.email)
+        return publisher
+
+    @staticmethod
+    def generate_engine(engineargs):
+        engine = Engine(engineargs.engine_name, engineargs.engine_version)
+        return engine
+
+    @staticmethod
+    def generate_model(modelargs):
+        model = Model(modelargs.model_name, modelargs.description, modelargs.model_version, dict(), modelargs.handler)
+        return model
+
+    @staticmethod
+    def generate_manifest_json(args):
+        """
+        Function to generate manifest as a json string from the inputs provided by the user in the command line
+        :param args:
+        :return:
+        """
+
+        publisher = ModelExportUtils.generate_publisher(args.publisher)
+        engine = ModelExportUtils.generate_engine(args.engine)
+        model = ModelExportUtils.generate_model(args.model)
+
+        try:
+            manifest = Manifest(args.runtime, engine, model, publisher, args.specification_version,
+                                args.implementation_version, args.model_server_version, args.license, args.description,
+                                args.user_data)
+
+        except ModelPackagingError as err:
```

Comment:
Why catch and rethrow?

Reply:
Removed it in ab6a918fdd4745c8ccaf884409a7393163a7ddb9